File: crates/normalize-languages/tests/fixtures/lua/sample.lua
local json = require("json")
local utils = require("utils.string")

-- Simple stack implementation
local Stack = {}
Stack.__index = Stack

function Stack.new()
    return setmetatable({ items = {}, size = 0 }, Stack)
end

function Stack:push(value)
    self.size = self.size + 1
    self.items[self.size] = value
end

function Stack:pop()
    if self.size == 0 then
        return nil
    end
    local value = self.items[self.size]
    self.items[self.size] = nil
    self.size = self.size - 1
    return value
end

function Stack:is_empty()
    return self.size == 0
end

-- Classify a number
function classify(n)
    if n < 0 then
        return "negative"
    elseif n == 0 then
        return "zero"
    else
        return "positive"
    end
end

-- Sum even numbers in a table
function sum_evens(tbl)
    local total = 0
    for _, v in ipairs(tbl) do
        if v % 2 == 0 then
            total = total + v
        end
    end
    return total
end

-- Count occurrences of each element
function count_occurrences(tbl)
    local counts = {}
    for _, v in ipairs(tbl) do
        counts[v] = (counts[v] or 0) + 1
    end
    return counts
end

local s = Stack.new()
s:push(1)
s:push(2)
print(classify(-3))
print(sum_evens({1, 2, 3, 4, 5}))

File: crates/normalize-languages/tests/fixtures/r/sample.r
library(stats)
library(utils)

# Classify a number as negative, zero, or positive
classify <- function(n) {
  if (n < 0) {
    return("negative")
  } else if (n == 0) {
    return("zero")
  } else {
    return("positive")
  }
}

# Sum even numbers in a vector
sum_evens <- function(values) {
  total <- 0
  for (v in values) {
    if (v %% 2 == 0) {
      total <- total + v
    }
  }
  return(total)
}

# Count occurrences of each unique value
count_occurrences <- function(values) {
  counts <- list()
  for (v in values) {
    key <- as.character(v)
    if (is.null(counts[[key]])) {
      counts[[key]] <- 1
    } else {
      counts[[key]] <- counts[[key]] + 1
    }
  }
  return(counts)
}

# Compute factorial recursively
factorial_r <- function(n) {
  if (n <= 1) {
    return(1)
  }
  return(n * factorial_r(n - 1))
}

print(classify(-3))
print(sum_evens(1:10))
print(factorial_r(5))
